gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/OrganizationCommandHandler.java

```diff
@@ -95,13 +98,17 @@ public Single<OrganizationReply> handle(OrganizationCommand command) {
             log.info("Organization [{}] handled with id [{}].", organization.getName(), organization.getId());
             return Single.just(new OrganizationReply(command.getId(), CommandStatus.SUCCEEDED));
         } catch (Exception e) {
-            log.error(
-                "Error occurred when handling organization [{}] with id [{}].",
-                organizationPayload.getName(),
-                organizationPayload.getId(),
-                e
-            );
+            log.error("Error occurred when handling organization [{}] with id [{}].", organizationPayload.getName(), organizationId, e);
             return Single.just(new OrganizationReply(command.getId(), CommandStatus.ERROR));
         }
     }
+
+    private String getOrganizationId(OrganizationPayload organizationPayload) {
+        try {
+            OrganizationEntity byCockpitId = this.organizationService.findByCockpitId(organizationPayload.getCockpitId());
+            return byCockpitId.getId();
+        } catch (Exception ex) {
```

Comment:
```suggestion
        } catch (OrganizationNotFoundException ex) {
```
I would only use the payload id if the organization is not found if a technical error occurs we could create a duplicate 🤔